### PR TITLE
[consensus] Get highest accepted rounds from RoundProber and use in ancestor selection

### PR DESCRIPTION
**Pull request by arun-koshy**
## Description 

For ancestor selection we want to ensure the highest quality ancestors are included in the proposal, therefore if an authority is excluded we should start including them once they have caught up to the network with their accepted blocks not just received as its possible for blocks to be received but suspended on many hosts which can cause performance degradation

Additionally switched to using high quorum round of the high quorum round of accepted rounds for inclusion metric.

Also bumped protocol version to 70 to include the new feature.

## Test plan 

How did you test the new or updated feature?

Injected Latency Test - https://metrics.sui.io/goto/Jowh4F7Hg?orgId=1
Hoarded Block Test - Pending

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [X] Protocol: v70 Enable probing for accepted rounds in round prober in consensus.
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-protocol-config/src/lib.rs
Comment (arun-koshy) [inline crates/sui-protocol-config/src/lib.rs:201]: Once Tasos lands his PR with Protocol Version 70 I will rebase this PR and move this feature to that version

File: consensus/core/src/round_prober.rs
Comment (arun-koshy) [inline consensus/core/src/round_prober.rs:269]: @mwtian can you just double check I am good to remove this, when I reviewed your PR initially I convinced myself it was needed but now I have convinced myself it is not.

**Review by arun-koshy — COMMENTED** [review]
The second commit using the high quorum round of the high quorum round is going to be tested but I can also move it to another PR if you want to review the changes separately.

File: consensus/core/build.rs
Comment (mwtian) [inline consensus/core/build.rs:89]: We intentionally used protobuf / prost for request & response of GetLatestRounds. Can you try adding to that instead?

File: consensus/core/build.rs
Comment (mwtian) [inline consensus/core/build.rs:159]: Since anemo is not used in prod, we can do incompatible changes as well.

File: consensus/core/src/round_prober.rs
Comment (mwtian) [reply consensus/core/src/round_prober.rs:269]: Probably a leftover from a previous impl. This shouldn't be needed anymore.

File: consensus/core/build.rs
Comment (arun-koshy) [reply consensus/core/build.rs]: Good call, updated!

File: consensus/core/build.rs
Comment (arun-koshy) [reply consensus/core/build.rs]: Looks much better now, thanks! Go protobuf 😄

File: consensus/core/src/core.rs
Comment (akichidis) [inline consensus/core/src/core.rs:273]: I think you can just to in the end:

```
let accepted_block_refs: BTreeSet<BlockRef> = accepted_blocks
                .iter()
                .map(|block| block.reference())
                .collect();
```

and return that. Probably not necessary to if-else here.

File: consensus/core/src/core.rs
Comment (mwtian) [inline consensus/core/src/core.rs:251]:
```suggestion
    /// - The references of accepted blocks
    /// - The references of ancestors missing their block
```

File: consensus/core/src/core_thread.rs
Comment (mwtian) [inline consensus/core/src/core_thread.rs:299]: I don't think we need to keep track of `highest_accepted_rounds` inside `ChannelCoreThreadDispatcher` and plumb accepted blocks back here. It seems whenever the `highest_accepted_rounds` is needed, it can be read with `dag_state.get_last_cached_block_per_authority(Round::MAX)...`

File: consensus/core/src/metrics.rs
Comment (mwtian) [inline consensus/core/src/metrics.rs:662]: Maybe update this to "Received round gaps among peers ..."

File: consensus/core/src/network/anemo_network.rs
Comment (mwtian) [inline consensus/core/src/network/anemo_network.rs:248]: Why we want to drop this conversion?

File: consensus/core/src/round_prober.rs
Comment (mwtian) [inline consensus/core/src/round_prober.rs:4]:
```suggestion
//! RoundProber periodically checks each peer for the latest rounds they received and accepted
```

File: consensus/core/src/round_prober.rs
Comment (mwtian) [inline consensus/core/src/round_prober.rs:13]:
```suggestion
//! by the CoreThreadDispatcher and DagState. The received rounds are updated after blocks are verified but before
```

File: consensus/core/src/round_prober.rs
Comment (mwtian) [inline consensus/core/src/round_prober.rs:12]:
```suggestion
//! RoundProber's data sources include the `highest_received_rounds` & `highest_accepted_rounds` tracked
```

File: consensus/core/src/round_prober.rs
Comment (mwtian) [inline consensus/core/src/round_prober.rs:180]: This condition seems not quite right. What if `consensus_round_prober_probe_accepted_rounds` is false and peers are returning empty `accepted`? Then `highest_accepted_rounds` for the `peer` will be set to empty and `compute_quorum_round()` will crash. It seems better to use separate if blocks for received and accepted.

**Review by mwtian — COMMENTED** [review]
Lgtm, with just a few comments.

File: consensus/core/src/round_prober.rs
Comment (arun-koshy) [reply consensus/core/src/round_prober.rs]: Great catch!! Thanks 🙇🏽

File: consensus/core/src/round_prober.rs
Comment (akichidis) [inline consensus/core/src/round_prober.rs:190]: Maybe we can add the error type as a label ? Ex `invalid_accepted_rounds` , `invalid_received_rounds`

File: consensus/core/src/core_thread.rs
Comment (akichidis) [inline consensus/core/src/core_thread.rs:103]: nit: instead of a tuple maybe you could use a `struct` instead as now there are quite a few parameters and would be good to have better clarity on the semantics.

**Review by akichidis — APPROVED** [review]
Looks good @arun-koshy ! I am approving but I would suggest for waiting for @mwtian to approve as well.

File: Cargo.lock
Comment (mwtian) [inline Cargo.lock:844]: Are these downgrades & changes intentional?

File: consensus/core/src/network/anemo_network.rs
Comment (mwtian) [inline consensus/core/src/network/anemo_network.rs:790]: Btw I think we can take the opportunity to change the types to `Round`.

**Review by mwtian — APPROVED** [review]

**Pull request merged**
